File: Android/scenes/ktv/src/main/java/io/agora/scene/ktv/widget/song/OnSongActionListener.java
package io.agora.scene.ktv.widget.song;

import androidx.annotation.NonNull;

/**
 * The interface On song action listener.
 */
public interface OnSongActionListener {

    /**
     * Order - callback on refresh
     *
     * @param dialog the dialog
     */
    void onChooseSongRefreshing(@NonNull SongDialog dialog);

    /**
     * Order - callback when the ‘select song’ button is clicked on an item.
     *
     * @param dialog   the dialog
     * @param songItem the song item
     */
    void onChooseSongChosen(@NonNull SongDialog dialog, @NonNull SongItem songItem);

    /**
     * Queued song list - callback when the ‘delete’ button on an item is clicked.
     *
     * @param dialog the dialog
     * @param song   the song
     */
    void onChosenSongDeleteClicked(@NonNull SongDialog dialog, @NonNull SongItem song);

    /**
     * Queued song list - callback when the ‘pin’ button on an item is clicked.
     *
     * @param dialog the dialog
     * @param song   the song
     */
    void onChosenSongTopClicked(@NonNull SongDialog dialog, @NonNull SongItem song);

}

File: Android/scenes/cantata/src/main/java/io/agora/scene/cantata/ui/widget/song/OnSongActionListener.kt
package io.agora.scene.cantata.ui.widget.song

interface OnSongActionListener {
    /**
     * 点歌-当下拉刷新时回调
     */
    fun onChooseSongRefreshing(dialog: SongDialog, index: Int)

    /**
     * 点歌-当加载更多时回调
     */
    fun onChooseSongLoadMore(dialog: SongDialog, index: Int)

    /**
     * 点歌-当点击搜索时回调
     */
    fun onChooseSongSearching(dialog: SongDialog, condition: String?)

    /**
     * 点歌-item"点歌"按钮点击时回调
     */
    fun onChooseSongChosen(dialog: SongDialog, songItem: SongItem)

    /**
     * 已点歌单-当item"删除"点击时回调
     */
    fun onChosenSongDeleteClicked(dialog: SongDialog, song: SongItem)

    /**
     * 已点歌单-当item"置顶"点击时回调
     */
    fun onChosenSongTopClicked(dialog: SongDialog, song: SongItem)
}